### PR TITLE
Instagram presence

### PR DESCRIPTION
**Pull request by MiepHD**
I added details for the nametag site, changed it so that it doesn't create a new presenceData everytime and fixed ESLint issues.

### CODE REVIEW

File: websites/I/Instagram/presence.ts [hunk 1]
```diff
@@ -1,61 +1,41 @@
...
-    const presenceData: PresenceData = {
-      details: "Viewing a profile",
-      state: document.location.pathname.split("/")[1],
-      largeImageKey: "logo"
-    };
+      presenceData.details = "Viewing a profile";
+      presenceData.state = document.location.pathname.split("/")[1];
+    }
+	
+	if (presenceData.details === null) {
```
Comment (ririxidev) [inline websites/I/Instagram/presence.ts:35]: from what I see - details is always a thing so maybe remove this if and leave only function from else :/

**Review by rxri — CHANGES_REQUESTED** [review]
Fix DeepScan & Codacy issues

**Comment by MiepHD** [comment]
I've fixed the issues except of "Use array destructuring" because I don't know how to fix it.

**Comment by ririxidev** [comment]
Instead of for example:
`const f = variable[0]`
you do:
`const [f] = variable`

if it's second or more you do:
`const [, f] = variable`

**Comment by MiepHD** [comment]
Like that?

File: websites/I/Instagram/presence.ts [hunk 2]
```diff
@@ -1,61 +1,35 @@
...
+  ) presenceData.details = "Direct Messages";
+    else {
     // TODO: Check if the page is really a profile
-    const presenceData: PresenceData = {
-      details: "Viewing a profile",
-      state: document.location.pathname.split("/")[1],
-      largeImageKey: "logo"
-    };
+      presenceData.details = "Viewing a profile";
+      presenceData.state = document.location.pathname.split("/")[, 1];
```
Comment (ririxidev) [inline websites/I/Instagram/presence.ts:31]:
```suggestion
      const [, pathName] = document.location.pathname.split("/")
      presenceData.details = "Viewing a profile";
      presenceData.state = pathName;
```

File: websites/I/Instagram/presence.ts [hunk 3]
```diff
@@ -1,61 +1,35 @@
...
-    presence.setActivity(presenceData);
+	const presenceData: PresenceData = {
+		largeImageKey: "logo"
+	};
+
+  if (document.location.pathname === "/") 
+      presenceData.details = "Viewing the homepage";
+	else if (document.location.pathname.startsWith("/stories")) {
+      presenceData.details = "Viewing a story";
+      presenceData.state = document.location.pathname.split("/")[, , 2];
```
Comment (ririxidev) [inline websites/I/Instagram/presence.ts:14]:
```suggestion
      const [, , pathName] = document.location.pathname.split("/")
      presenceData.details = "Viewing a story";
      presenceData.state = pathName;
```

**Review by rxri — CHANGES_REQUESTED** [review]
like this

File: websites/I/Instagram/presence.ts (hunk 3 above)
Comment (MiepHD) [reply websites/I/Instagram/presence.ts:14]: Thanks now I understand how to solve it

File: websites/I/Instagram/dist/metadata.json [hunk 4]
```diff
@@ -16,7 +16,7 @@
     "zh_TW": "Instagram是Facebook公司旗下一款免費提供線上圖片及視訊分享的社群應用軟體。"
   },
   "url": "www.instagram.com",
-  "version": "2.0.2",
+  "version": "2.1.0",
   "logo": "https://crugg.de/premid/iglogo.png",
   "thumbnail": "https://crugg.de/premid/igbanner.png",
```
Comment (ririxidev) [inline websites/I/Instagram/dist/metadata.json:21]: upload to imgur

File: websites/I/Instagram/presence.ts [hunk 5]
```diff
@@ -1,61 +1,37 @@
...
-    };
-    presence.setActivity(presenceData);
+	const presenceData: PresenceData = {
+		largeImageKey: "logo"
+	};
+
+  if (document.location.pathname === "/") 
+      presenceData.details = "Viewing the homepage";
+	else if (document.location.pathname.startsWith("/stories")) {
+      const [, , pathName] = document.location.pathname.split("/")
```
Comment (ririxidev) [inline websites/I/Instagram/presence.ts:13]:
```suggestion
      const [, , pathName] = document.location.pathname.split("/");
```

File: websites/I/Instagram/presence.ts [hunk 6]
```diff
@@ -1,61 +1,37 @@
...
-  } else {
+  ) presenceData.details = "Direct Messages";
+    else {
     // TODO: Check if the page is really a profile
-    const presenceData: PresenceData = {
-      details: "Viewing a profile",
-      state: document.location.pathname.split("/")[1],
-      largeImageKey: "logo"
-    };
+      const [, pathName] = document.location.pathname.split("/")
```
Comment (ririxidev) [inline websites/I/Instagram/presence.ts:31]:
```suggestion
      const [, pathName] = document.location.pathname.split("/");
```

**Review by rxri — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]
LGTM

**Review by rxri — APPROVED** [review]

**Pull request merged**